src/xcode/ENA/ENA/Source/Scenes/DynamicTableViewController/Views/DynamicTableViewRoundedCell.swift

```diff
@@ -85,16 +85,20 @@ class DynamicTableViewRoundedCell: UITableViewCell {
 	private func setupConstraints() {
 		let marginGuide = contentView.layoutMarginsGuide
 		contentView.addSubview(insetView)
-
+		
+		// Need top opt-out the constraints to avoid empty spaces if title or body are not set but the button.
+		let titleTopConstraintValue: CGFloat = title.attributedText == nil ? 0 : 20
+		let bodyTopConstraintValue: CGFloat = body.attributedText == nil ? 0 : 16
+		
 		NSLayoutConstraint.activate([
 			insetView.leadingAnchor.constraint(equalTo: marginGuide.leadingAnchor),
 			insetView.topAnchor.constraint(equalTo: marginGuide.topAnchor),
 			insetView.trailingAnchor.constraint(equalTo: marginGuide.trailingAnchor),
 			insetView.bottomAnchor.constraint(equalTo: marginGuide.bottomAnchor),
 			title.leadingAnchor.constraint(equalTo: insetView.leadingAnchor, constant: 16),
-			title.topAnchor.constraint(equalTo: insetView.topAnchor, constant: 20),
+			title.topAnchor.constraint(equalTo: insetView.topAnchor, constant: titleTopConstraintValue),
```

Comment:
Will this affect other cells besides the settings screen?

Reply:
No, the others screen using the IconCell never pass a nil text and body.